File: Q23/Q23.go
package main

import (
	"fmt"
)

func printBinaryCheckerboard(n int) {
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			fmt.Printf("%d ", (i+j)%2)
		}
		fmt.Println()
	}
}

func main() {
	var size int
	fmt.Print("Enter the size of the checkerboard: ")
	fmt.Scan(&size)

	printBinaryCheckerboard(size)
}

File: Q23/Q23.cpp
#include <iostream>
using namespace std;

void printBinaryCheckerboard(int n) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << (i + j) % 2 << " ";
        }
        cout << endl;
    }
}

int main() {
    int size;
    cout << "Enter the size of the checkerboard: ";
    cin >> size;

    printBinaryCheckerboard(size);

    return 0;
}
